src/averageAges.js

```diff
@@ -55,7 +69,19 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = onlyWithSon
+    ? people.filter(p => p.sex === 'm'
+      && people.some(m => m.name === p.mother))
+    : people.filter(p => people.some(m => m.name
+      === p.mother));
+
+  let sum = 0;
+
+  children.forEach((ch) => sum += ch.born - people.find(
+    (m) => m.name === ch.mother
+  ).born);
```

Comment:
According the above comments about use of reduce() function for calculate a single value and having stored the child's mom before, you can replace this block with:

```suggestion
  const sum = children.reduce(
    (accumulator, child) => accumulator + (child.born - child.mom.born),
    0
  );
```